src/handshake/crypto.js

```diff
@@ -77,6 +78,16 @@ exports.identify = (state, msg, callback) => {
     // If we know who we are dialing to, double check
     if (state.id.remote) {
       if (state.id.remote.toB58String() !== remoteId.toB58String()) {
+        if (pubkey.length > 42) {
+          return callback(new Error('dialed to the wrong peer, Ids do not match'))
+        }
+
+        // This peer ID may have been calculated with identity hash
+        const idPeerId = mh.encode(pubkey, 'identity')
+        if (state.id.remote.toB58String() !== mh.toB58String(idPeerId)) {
+          return callback(new Error('dialed to the wrong peer, Ids do not match'))
+        }
+
         return callback(new Error('dialed to the wrong peer, Ids do not match'))
```

Comment:
There seems to be an issue with the logic here unless I am missing something. 

Regardless of what happens, the callback is always going to get called with the `Ids do not match` error. Does line 91 need to get removed?

Reply:
@jacobheun Ow, you're right :/ I wonder how the code ended up in this state. I'm sure I got this working, maybe it was a different revision.

Reply:
Anyway, a test will ensure that the logic is working.